src/rest_vol_dataset.c

```
@@ -5217,6 +5369,370 @@ RV_get_cmpd_subset_nmembers(hid_t src_type_id, hid_t dst_type_id, size_t *num_cm
+
+    in = (const hvl_t *)_in;
+
+    if ((parent_type = H5Tget_super(dtype_id)) < 0)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_BADVALUE, FAIL, "can't get base type of vlen type");
+
+    if ((parent_class = H5Tget_class(parent_type)) < 0)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_CANTGET, FAIL, "can't get class of parent of vlen dtype");
+
+    if ((parent_dtype_size = H5Tget_size(parent_type)) == 0)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_CANTGET, FAIL, "can't get size of vlen seq base type");
+
+    if ((out_string = (char *)calloc(DATASET_VLEN_JSON_BODY_DEFAULT_SIZE, 1)) == NULL)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_CANTALLOC, FAIL, "can't allocate space for vlen data JSON");
+
+    *out_size           = DATASET_CREATION_PROPERTIES_BODY_DEFAULT_SIZE;
+    out_string_curr_pos = out_string;
+
+    /* Delineate start of list of sequences */
+    out_diff     = out_string_curr_pos - out_string;
+    tgt_out_size = (size_t)out_diff;
+    CHECKED_REALLOC(out_string, *out_size, tgt_out_size, out_string_curr_pos, H5E_DATASET, FAIL);
+    out_string = strcat(out_string, "[");
+    out_string_curr_pos += 1;
+
+    /* The amount of space needed to serialize each element can't be known ahead of time, so serialize each
+     * element into a temporary fixed-size buffer before copying it to the output buffer . */
+    for (size_t i = 0; i < nelems; i++) {
+        hvl_t seq = in[i];
+
+        /* Delineate start of this sequence */
+        out_diff     = out_string_curr_pos - out_string;
+        tgt_out_size = (size_t)(out_diff) + 1;
+        CHECKED_REALLOC(out_string, *out_size, tgt_out_size, out_string_curr_pos, H5E_DATASET, FAIL);
+        out_string = strcat(out_string, "[");
+        out_string_curr_pos += 1;
+
+        /* Stringify each element in a second fixed-size buffer, then copy to sequence buffer */
+        for (size_t j = 0; j < seq.len; j++) {
+#define MAX_SEQ_ELEM_BUF_SIZE 128
```

We should generally avoid inline `#define`s anywhere and keep them either at the top of the file or at the very least right above the function they're used in. That said, it doesn't seem like we necessarily need to have a fixed limit on the number of sequence elements we support, but if so, we should at least throw an error if a sequence is too large.